utils/base_tests.py

```diff
@@ -0,0 +1,164 @@
+        self.__field_input_acceptance_test(
+            field=field,
+            value=value,
+            is_valid=False,
+            error_class=None,
+            error_message=None,
+        )
+
+    """
+    Test that putting specified value in specified field should not result in successful model validation.
+    An error class and it's message can be additionally specified for TestCase assertion.
+    Specifying error_message results in calling assertRaisesRegexp(), comparing it with received message.
+    Leaving error_message empty will result in calling assertRaises().
+    Leaving error_class empty will result in assertion for ValidationError by default.
+    """
+    def field_should_not_accept_input(self, field, value, error_class=None, error_message=None):
+        self.__field_input_acceptance_test(
+            field=field,
+            value=value,
+            is_valid=False,
+            error_class=error_class,
+            error_message=error_message,
+        )
+
+    """
+    Test that leaving specified value empty should result in successful model validation.
+    """
+    def field_should_accept_null(self, field):
+        self.field_should_accept_input(field, None)
+
+    """
+    Test that putting specified value in specified field should not result in successful model validation.
+    Support for specifying error class and error message is preserved in case a special behavior is expected.
+    """
+    def field_should_not_accept_null(self, field, error_class=None, error_message=None):
+        self.field_should_not_accept_input(
+            field=field,
+            value=None,
+            error_class=error_class,
+            error_message=error_message,
+        )
+
+    """
+    Test that putting specified value in specified field should result in storing that value in database.
+    """
+    def data_should_be_stored(self, field, value=None):
+        model = self.save_and_load_model_from_database(self.initiate_model(field, value))
+        self.assertEqual(getattr(model, field), value)
+
+    """
+    Test that timestamp field with auto_now flag set to true should change on model update.
+        auto_field -> field with auto_now flag to be tested
+        update_field -> field that should be updated for test's purpose
+        value -> correct input for field that should be updated
+    WARNING: It is assumed that the field is of DateTime type.
+    Using any other time-related type can result in fail!
+    """
+    def field_auto_now_test(self, auto_field, update_field, value):
+        model = self.default_model()
+        model.full_clean()
+        model.save()
+        previous_value = getattr(model, auto_field)
+        model_pk = model.pk
+        model = self.model_class.objects.get(pk=model_pk)
+        setattr(model, update_field, value)
+        model.full_clean()
+        model.save()
+        current_value = getattr(model, auto_field)
+        self.assertNotEqual(previous_value, current_value)
```

Comment:
It would be safer if you use mocking datetimes to do that - it is a possibility that values will not differ at all.
Bu this is improvement to add to Pivotal.